### PR TITLE
feat(albums): Adds album of images to AC notification

### PR DESCRIPTION
**Pull request by borismelnik**
This PR adds album of images to AC notification which needs for displaying the images in AC
[desktop issue](https://github.com/status-im/status-desktop/issues/9951)

Important changes:
- [x] Something worth noting for reviewers.

Closes #

### CODE REVIEW

**Review by MishkaRogachev — APPROVED** [review]
LGTM

**Review by jrainville — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Comment by borismelnik** [comment]
@ilmotta @MishkaRogachev After discussion with @jrainville we decide to add album to AC notification structure

File: protocol/activity_center_persistence.go [hunk 1]
```diff
@@ -307,6 +307,16 @@ func (db sqlitePersistence) unmarshalActivityCenterNotificationRow(row *sql.Row)
 		if err = json.Unmarshal(messageBytes, message); err != nil {
 			return nil, err
 		}
+		image := message.GetImage()
+		if image != nil {
+			if image.AlbumId != "" {
```
Comment (jrainville) [inline protocol/activity_center_persistence.go]:
```suggestion
		if image != nil && image.AlbumId != "" {
```

File: protocol/activity_center_persistence.go [hunk 2]
```diff
@@ -387,6 +397,16 @@ func (db sqlitePersistence) unmarshalActivityCenterNotificationRows(rows *sql.Ro
 			if err = json.Unmarshal(messageBytes, message); err != nil {
 				return "", nil, err
 			}
+			image := message.GetImage()
+			if image != nil {
+				if image.AlbumId != "" {
```
Comment (jrainville) [inline protocol/activity_center_persistence.go]:
```suggestion
			if image != nil && image.AlbumId != "" {
```

**Review by jrainville — APPROVED** [review]
LGTM

File: protocol/messenger.go [hunk 3]
```diff
@@ -3313,25 +3313,38 @@ func (r *ReceivedMessageState) addNewActivityCenterNotification(publicKey ecdsa.
 	var idToUse string
```
Comment (MishkaRogachev) [inline protocol/messenger.go]: notificationID will be more obvious

**Review by MishkaRogachev — APPROVED** [review]

File: protocol/messenger.go (hunk 3 above)
Comment (borismelnik) [reply protocol/messenger.go]: I don't understand its not my code :)

Comment (borismelnik) [reply protocol/messenger.go]: I know that code was added by @jrainville if he don't mind I can fix :)

Comment (jrainville) [reply protocol/messenger.go]: You can change it

**Comment by borismelnik** [comment]
@Samyoul @cammellos Could you pls help me to understand what's wrong here? the output gives me nothing

**Comment by cammellos** [comment]
@borismelnik 
```
[2023-09-13T00:39:12.604Z] panic: test timed out after 30m0s
```
Unfortunately tests are taking quite a long time, we should probably start thinking about either breaking them up, etc.

on that note, could you please add a test to this PR validating the behavior please?

**Comment by borismelnik** [comment]
> @borismelnik
> 
> ```
> [2023-09-13T00:39:12.604Z] panic: test timed out after 30m0s
> ```
> 
> Unfortunately tests are taking quite a long time, we should probably start thinking about either breaking them up, etc.
> 
> on that note, could you please add a test to this PR validating the behavior please?

yeah sure :) so I just need to restart and restart it till test will be passed?

**Comment by cammellos** [comment]
@borismelnik for now yes, and write a new one as well, just to add insult to injury :)

**Comment by Samyoul** [comment]
Weird error in https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-3977/29/pipeline `Failed, see the log`

```
[2023-09-19T09:58:52.468Z] Testing: github.com/status-im/status-go/protocol
[2023-09-19T10:29:14.798Z] Failed, see the log: /home/jenkins/workspace/status-go_prs_tests_PR-3977/protocol/test.log
[2023-09-19T10:29:14.798Z] make: *** [Makefile:319: test-unit] Error 1
```

Screenshot 2023-09-19 at 13 47 46

Restarting tests

**Comment by borismelnik** [comment]
> Weird error in https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-3977/29/pipeline `Failed, see the log`
> 
> ```
> [2023-09-19T09:58:52.468Z] Testing: github.com/status-im/status-go/protocol
> [2023-09-19T10:29:14.798Z] Failed, see the log: /home/jenkins/workspace/status-go_prs_tests_PR-3977/protocol/test.log
> [2023-09-19T10:29:14.798Z] make: *** [Makefile:319: test-unit] Error 1
> ```
> 
> Screenshot 2023-09-19 at 13 47 46
> Restarting tests

yeah I see the same always, but locally all tests passed

**Comment by borismelnik** [comment]
@Samyoul its always takes 33-36 mins

**Comment by Samyoul** [comment]
I'm running all tests in `/protocol` locally, So far stuck for over 30 minutes on:

```go
func TestMessengerActivityCenterMessageSuite(t *testing.T) {
	suite.Run(t, new(MessengerActivityCenterMessageSuite))
}
```

```go
func (s *MessengerActivityCenterMessageSuite) TestReplyWithImage() {
```

All of the other tests in the `MessengerActivityCenterMessageSuite` completed a while ago.

Screenshot 2023-09-19 at 14 37 01

**Pull request merged**